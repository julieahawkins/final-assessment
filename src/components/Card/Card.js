import React from 'react';
import PropTypes, { object } from 'prop-types';

const Card = ({ houseData }) => {
  const { 
    name, 
    words, 
    founded, 
    seats, 
    titles, 
    ancestralWeapons, 
    coatOfArms 
  } = houseData;

  const weaponsDisplay = ancestralWeapons.map((weapon, index) => 
    <p key={`weapon-${index}`}>Ancestral Weapons: {weapon}</p>
  );

  const seatDisplay = seats.map((seat, index) => 
    <p key={`seat-${index}`}>Seat: {seat}</p>
  );

  const titleDisplay = titles.map((title, index) => 
    <p key={`title-${index}`}>Title: {title}</p>
  );

  return (
    <div className='Card'>
      <h1>{name}</h1>
      <h2>{words}</h2>
      <h3>Founded: {founded || 'N/A'}</h3>
      {seatDisplay}
      {titleDisplay}
      {weaponsDisplay}
      <p>Coat of Arms: {coatOfArms}</p>
    </div>
  );
};

Card.propTypes = {
  houseData: object.isRequired
};

export default Card;