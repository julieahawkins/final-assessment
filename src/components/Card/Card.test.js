import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

describe('Card test', () => {
  const fakeStore = configureMockStore()({
    currentSwornData: []
  });


  const setup = () => {
    const props = {
      showMoreDetails: jest.fn()
    };

    const mockHouseDataProps = {
      name: 'House Hightower of the Hightower', 
      words: 'We Light the Way', 
      founded: 'Age of Dawn', 
      seats: ['The Hightower, Oldtown'], 
      titles: [
        'Lord of the Hightower',
        'Lord of the Port',
        'Voice of Oldtown',
        'Defender of the Citadel',
        'Beacon of the South',
        'King of the High Tower (formerly)'
      ], 
      ancestralWeapons: ['Vigilance'], 
      coatOfArms: 'Cendr√©e, a tower argent with a beacon on fire gules'  
    };

    const wrapper = shallow(
      <Provider store={fakeStore}>
        <Card houseData={mockHouseDataProps}/>
      </Provider>
    );
    const Component = wrapper.find(Card);

    return {
      Component
    };
  };

  it('should match its snap shot', () => {
    
    expect(setup.wrapper).toMatchSnapshot();
  });
});