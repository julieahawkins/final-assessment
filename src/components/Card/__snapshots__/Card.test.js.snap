// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test should match its snap shot 1`] = `undefined`;
